### PR TITLE
test case when xts mode key and tweak init is done separately

### PR DESCRIPTION
**Pull request by torben-hansen**
### Issues:

V874784593

### Description of changes: 

https://github.com/aws/aws-lc/pull/963 implemented some test cases for XTS usage models. I forgot about this usage model where you init the key and first tweak separately. This better aligns with how one would to this in an actual application.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and 
the ISC license.

### CODE REVIEW

File: crypto/fipsmodule/modes/xts_test.cc [hunk 1]
```diff
@@ -857,11 +865,25 @@ TEST(XTSTest, SectorTweakAPIUsage) {
...
     nullptr, nullptr, sectorTweak2.data()));
   ASSERT_TRUE(EVP_DecryptInit_ex(ctx_decrypt.get(), nullptr,
     nullptr, nullptr, sectorTweak2.data()));
-  encrypt_and_decrypt(ctx_encrypt, ctx_decrypt, pt, ct2_expected);
+  encrypt_and_decrypt(ctx_encrypt, ctx_decrypt, pt, ct2_expected, false, {});
+
+  ctx_encrypt.Reset();
+  ctx_decrypt.Reset();
+
+  // Finally, encrypt and decrypt but only re-init the sector tweak and do the
```
[the remark this answers is not in the document]
Comment (torben-hansen) [reply crypto/fipsmodule/modes/xts_test.cc:879]: >   // Thirdly, encrypt and decrypt but only re-init the sector tweak.
  
I think the second part of the sentence highlights the difference between the two. I think.

>  // Finally, encrypt and decrypt but only re-init the sector tweak and do the
// key init and sector tweak init in different function calls.

[the remark this answers is not in the document]
Comment (torben-hansen) [reply crypto/fipsmodule/modes/xts_test.cc:879]: The input changes:
```C
encrypt_and_decrypt(ctx_encrypt, ctx_decrypt, pt, ct1_expected, true, sectorTweak1);
encrypt_and_decrypt(ctx_encrypt, ctx_decrypt, pt, ct2_expected, true, sectorTweak2);
```

[the remark this answers is not in the document]
Comment (nebeid) [reply crypto/fipsmodule/modes/xts_test.cc:879]: Ah, thank you for clarifying what I missed. I'll approve while suggesting changing the order of the sentence
```suggestion
  // Finally, encrypt and decrypt but do the key init and sector tweak init in different function calls
  // and only re-init the sector tweak
```

**Review by nebeid — APPROVED** [review]

**Review by nebeid — APPROVED** [review]

**Review by skmcgrail — APPROVED** [review]

File: crypto/fipsmodule/modes/xts_test.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (torben-hansen) [reply crypto/fipsmodule/modes/xts_test.cc]: Modified language to make it more sane.

**Pull request merged**